### PR TITLE
feat(TimePicker): new component

### PR DESCRIPTION
**Pull request by alberthovhannisyan1**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=1558 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-zazptlzbtu.chromatic.com/

### CODE REVIEW

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+// Components
+import Button from "@components/atoms/Button";
+import HelperText from "@components/atoms/HelperText";
+import { TimePickerSizes } from "@components/molecules/TimePicker/types";
+
+// Styles
+import "./PickerInput.scss";
+
+type InputMaskProps = ComponentProps<typeof InputMask>;
+type PickerInputOnClick = NonNullable<InputMaskProps["onClick"]>;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:17]: Why do we need this?

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+     */
+    EndIcon?: FC<IconProps>;
+    /**
+     * Callback function which triggers when the user clicks on a clear button.
+     */
+    handleClear?: () => void;
+    /**
+     * Condition when the clear button should be visible.
+     */
+    shouldShowClearableIcon?: boolean;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:66]: Maybe `isClearable` prop name is more descriptive?

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+     */
+    popoverRefData?: HTMLAttributes<HTMLDivElement>;
+    /**
+     * Used for accessibility, determines whether the picker popover is open/expanded.
+     */
+    isExpanded: boolean;
+    /**
+     * Used for accessibility, receives the id of the picker popover.
+     */
+    ariaControls?: string;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:78]: Do we need this ?

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+     */
+    value?: string | null;
+    /**
+     * ID used to tie the input field to a label.
+     */
+    id: string;
+    /**
+     * Aria-label attribute of the picker input field.
+     */
+    ariaLabel?: string;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:169]: Check all aria-* props, make sure whatever is added there is a need for that, also check for missing aria-*s

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+     * The values of the range input picker.
+     */
+    value?: {
+        start?: string | null;
+        end?: string | null;
+    };
+    /**
+     * IDs used to tie the input field to the label
+     */
+    ids: {
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:190]: Verify that this is required for the logic

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+    ariaControls = "Time picker shell"
+}) => {
+    const shouldShowIconAppends = shouldShowClearableIcon || EndIcon;
+    const helperTextSize = size === "medium" || size === "large" ? "medium" : "small";
+
+    return (
+        <>
+            <div
+                role="combobox"
+                aria-expanded={isExpanded}
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:226]: These props should be set not for container but for the specific element

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+                aria-controls={ariaControls}
+                className={classNames("pickerInput", className, `pickerInput_mode_${mode}`, {
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:228]: Move this upper into a constant

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 8]
```diff
@@ -0,0 +1,415 @@
...
+    EndIcon,
+    errorMessage,
+    onClear,
+    clearable,
+    value,
+    placeholder,
+    onClick,
+    onChange,
+    onFocus,
+    mask = "__:__:__",
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:280]: Move default mask into constants

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 9]
```diff
@@ -0,0 +1,415 @@
...
+            handleClear={onClear}
+            shouldShowClearableIcon={!!shouldShowClearableIcon}
+            popoverRefData={popoverRefData}
+            isExpanded={isExpanded}
+            ariaControls={ariaControls}
+        >
+            <InputMask
+                id={id}
+                mask={is12Hour ? "__:__:__ aa" : mask}
+                replacement={is12Hour ? { _: /[0-9]/, a: /[a-zA-Z]/ } : { _: /[0-9]/ }}
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:308]: Additionally verify the regexp and move it into constants

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx [hunk 10]
```diff
@@ -0,0 +1,415 @@
...
+                separate
+                onClick={() => onClick("start")}
+                onChange={(e) => onChange(e, "start")}
+                onFocus={onFocus}
+                aria-label={ariaLabels?.start || "Range time picker start time"}
+            />
+            <Minus className="pickerInput__icon" size={16} aria-hidden="true" />
+            <InputMask
+                id={ids?.end}
+                mask={is12Hour ? "__:__:__ aa" : mask}
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:390]: The part which is responsible for AM/PM is accepting other string values also, need to fix

File: src/components/molecules/TimePicker/components/PickerPopover/PickerPopover.tsx [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+    size?: TimePickerSizes;
+    /**
+     * Reference to the popover element.
+     */
+    popoverRef?: Ref<IPopoverRef>;
+    /**
+     * Popover positioning props passed to the underlying popover component.
+     */
+    setProps: Dispatch<SetStateAction<Record<string, unknown>>>;
+    onSelect?: (column: keyof TimeParts, val: string) => void;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerPopover/PickerPopover.tsx:44]: Add missing tsdoc

File: src/components/molecules/TimePicker/components/PickerPopover/PickerPopover.tsx [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+    };
+    /**
+     * Current active field
+     */
+    activeField?: "start" | "end";
+    /**
+     * Time parts.
+     */
+    partsStart?: TimeParts;
+    partsEnd?: TimeParts;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerPopover/PickerPopover.tsx:70]: Check for all props tsdoc comments

File: src/components/molecules/TimePicker/components/PickerPopover/PickerPopover.tsx [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+                                <div className="timePicker__header">
+                                    <Text className="ellipsis-text" as="p" variant={headerTextVariant}>
+                                        {text}
+                                    </Text>
+                                </div>
+                            </div>
+                            <div className="timePicker__body">
+                                <Scrollbar>
+                                    <div className="timePicker__list" role="presentation">
+                                        {data.map((item) => (
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/components/PickerPopover/PickerPopover.tsx:154]: Is this well optimized ??

File: src/components/molecules/TimePicker/TimePicker.test.tsx [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+    it("switches active popover target depending on clicked input", () => {
+        const inputs = setup.find("input.pickerInput__input");
+
+        inputs.at(0).simulate("click");
+
+        inputs.at(1).simulate("click");
+
+        expect(setup.find("PickerPopover").prop("open")).toBe(true);
+    });
+});
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/TimePicker.test.tsx:191]: Add more tests to handle edge cases

File: src/components/molecules/TimePicker/TimePicker.tsx [hunk 15]
```diff
@@ -0,0 +1,378 @@
...
+     * Callback function that is triggered when the popover is toggled.
+     * @param open
+     */
+    onPopoverToggle?: (open: boolean) => void;
+    /**
+     * Custom texts for the component.
+     * @param {string} texts.amText - The text to display for AM (AM/A).
+     * @param {string} texts.pmText - The text to display for PM (P/P).
+     */
+    texts?: {
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/TimePicker.tsx:90]: Verify the prop names, discuss with AI

File: src/components/molecules/TimePicker/TimePicker.tsx [hunk 16]
```diff
@@ -0,0 +1,378 @@
...
+        hours?: string;
+        minutes?: string;
+        seconds?: string;
+    };
+    /**
+     * Disabled specific time programmatically.
+     * @param type
+     * @param value
+     */
+    shouldDisableTime?: (type: "hours" | "minutes" | "seconds" | "meridiem", value: string) => boolean;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/TimePicker/TimePicker.tsx:102]: There is already defined type which can be reused here

File: src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx (hunk 2 above)
Comment (alberthovhannisyan1) [reply src/components/molecules/TimePicker/components/PickerInput/PickerInput.tsx:66]: UPD: this is the prop for shell which is not accessible for user, and the picker input itself is receiving prop `clearable` so it's correct to keep this
